### PR TITLE
Add utility for constructing a merkle proof against TreeNode.

### PR DESCRIPTION
**Pull request by robzajac**
## PR Description
Adds a utility class to construct the merkle proof for a field (represented by generalized index) in a given `TreeNode`.

Will shortly be used for verifying inclusion of `current_sync_committee` in the `BeaconState` for the `GetLightClientBootstrap` endpoint as per the [light client bootstrap spec](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/full-node.md#create_light_client_bootstrap).

See also [here](https://github.com/ethereum/consensus-specs/blob/dev/ssz/merkle-proofs.md#merkle-multiproofs) for more on merkle proofs and the `consensus-specs` implementations [setup.py](https://github.com/ethereum/consensus-specs/blob/dev/setup.py#L522) [merkle.py](https://github.com/ethereum/consensus-specs/blob/dev/tests/core/pyspec/eth2spec/test/helpers/merkle.py#L4)

## Fixed Issue(s)
partially addresses #4230 

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/tree/MerkleUtil.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    return Collections.unmodifiableList(proof);
+  }
+
+  /**
+   * Returns the path from the root to the specified node as an array of generalized indices.
+   * path[0] is the root itself (gIndex of 1) path[path.length - 1] is the node (gIndex of
+   * `nodeIndex`)
+   */
+  @VisibleForTesting
+  static List<Long> getPathToNode(long nodeIndex) {
```
Comment (rolfyone) [inline infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/tree/MerkleUtil.java:68]: nit: do these map to functions in the spec? The tests are good, just wondering if there's an algorithm I'm checking them against, and if we can make that relationship a little clearer if they're just a spec function by naming them in a way we can relate them directly...

Comment (robzajac) [reply infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/tree/MerkleUtil.java:68]: The spec doesn't define these functions from what I can see. The light client spec defines a [`compute_merkle_proof_for_state`](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/full-node.md#compute_merkle_proof_for_state) whose implementation is not provided. The impl in this PR is more general purpose than just an interface accepting `BeaconState`. The [`Merkle multiproofs`](https://github.com/ethereum/consensus-specs/blob/dev/ssz/merkle-proofs.md#merkle-multiproofs) spec does not provide these functions either. The only definitions I could see came from the test cases (linked in OP).

Comment (rolfyone) [reply infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/tree/MerkleUtil.java:68]: ok all good, just makes it hard to verify against a spec :)

**Review by rolfyone — APPROVED** [review]
LGTM as a starting point.

**Comment by robzajac** [comment]
Thanks @rolfyone do you mind merging in as I don't have write access? I'll followup with a PR integrating this into `GetLightClientBootstrap`

**Pull request merged**